Compression

jleiandy opened the pull request:
Added optimizations, new server endpoints, relocated static image assets, and refactored thumbnail inputs into divs. Please see commit comments for more details.

jleiandy committed:
Added compression to bundle.js

jleiandy committed:
Included brotli compressions

jleiandy committed:
Added code splitting.

cschucode committed:
bundles

cschucode committed:
Update index.html for SEO

jleiandy committed:
Added compression and code splitting optimizations
Added meta tags to HTML and refactored script elements to call for brotli files
Added separate endpoints to serve html and brotli files.
Moved static images to new folder, which are being served via express.static, will decide if need to make separate endpoints for images later.
Added transition attribute to imageGalleryContainer to smooth out image zoom mouse over.

jleiandy committed:
Refactored thumbnail and styleselector input elements into divs with
role of "button"
No longer need transparent image layer and allows for ease of access
to manage image properties via setting the image as the background of div and applying background css.

cschucode commented on client/components/ProductOverview/StyleSelector.jsx:
```diff
@@ -16,9 +16,13 @@ const StyleSelector = ({ styles, currentStyle, updateCurrentStyle }) => {
-                src="https://upload.wikimedia.org/wikipedia/commons/thumb/8/89/HD_transparent_picture.png/1200px-HD_transparent_picture.png"
+              <div
+                role="button"
+                onKeyPress={(e) => {
```
This works for now, but ideally it should be in a handler

jleiandy replied:
created handler

cschucode commented on client/components/ProductOverview/ThumbnailView.jsx:
```diff
@@ -57,14 +57,18 @@ const ThumbnailView = ({ photos, handleClick, currentIndex }) => {
+                }}
                 id={index === currentIndex ? 'selected' : 'unselected'}
                 key={photo.thumbnail_url}
-                src="https://upload.wikimedia.org/wikipedia/commons/thumb/8/89/HD_transparent_picture.png/1200px-HD_transparent_picture.png"
```
❤️

cschucode commented on package.json:
```diff
@@ -62,6 +63,7 @@
     "style-loader": "^3.2.1",
-    "styled-components": "^5.3.1"
+    "styled-components": "^5.3.1",
+    "zlib": "^1.0.5"
```
I was under the impression that 'zlib' was included with Node in versions after 10.16. (https://nodejs.org/api/zlib.html)  Do we need to import this?

jleiandy replied:
good catch, i uninstalled the zlib node module and our webpack is still working.

cschucode commented on server/index.js:
```diff
@@ -1,20 +1,58 @@
 const express = require('express');
 const path = require('path');
 const axios = require('axios');
+const { join } = require('path');
```
It looks like you import path on line 2. Any reason just to restructure join here? Or, do we not need the path import anymore?

jleiandy replied:
was just sloppy. refactored.

cschucode commented on client/css/ImageGallery.scss:
```diff
@@ -26,8 +28,7 @@
   background-size: 100%;
-  height: 100%;
-  width: 100%;
+  animation: flexanimation 5s ease-out;
```
I got to learn about this.

cschucode reviewed (approved):
Let's delete the webpack generated files and add them to the gitignore.

jleiandy committed:
Destructured only join method from path module
Added fixed height of 600px to main image to prevent it from collapsing when expanded
Added onkeypress handler functions to StyleSelector and ThumbnailView
Uninstalled superfluous zlib module
Added webpack bundles to gitignore.

jleiandy committed:
Delete main.js

jleiandy committed:
Delete main.js.br

jleiandy committed:
Delete vendor.js

jleiandy committed:
Delete vendor.js.LICENSE.txt

jleiandy committed:
Delete vendor.js.br

jleiandy committed:
Merge branch 'main' into compression

jleiandy committed:
Merge branch 'compression' of https://github.com/HRSF137-FEC-Enoki/catwalk into compression
